### PR TITLE
resizeCanvas integrated

### PR DESCRIPTION
**Pull request by BernardTeske**
Hi! I found your wonderful tool! It helped me a lot, but I needed one more thing. I need squared images (with same height and with) from originals that have an other ratio. So i integrated resizeCanvas. If you like take it to your repo.

### CODE REVIEW

**Comment by BernardTeske** [comment]
Oh yes! Sorry. It was my first pull request. And I am not really expert in composer.

**Comment by mirko-pagliai** [comment]
Now I see what I can do.
However, it is better if you do the PR in the future from a branch different from master (eg. `resize-canvas-fix`)

**Comment by BernardTeske** [comment]
Ok, thank you! I will do so in future!

File: src/Utility/ThumbCreator.php [hunk 1]
```diff
@@ -210,6 +210,32 @@ public function resize($width = null, $heigth = null, array $options = [])
...
+     * @param array $options Options for the thumbnail
+     * @return \Thumber\Utility\ThumbCreator
+     * @see https://github.com/mirko-pagliai/cakephp-thumber/wiki/How-to-uses-the-ThumbCreator-utility#resizeCanvas
+     * @uses $arguments
+     * @uses $callbacks
+     */
+    public function resizeCanvas($width = null, $heigth = null, array $options = [])
+    {
+        //Sets default options
+        $options += ['anchor' => 'center', 'relative' => 'false', 'bgcolor' => 'ffffff'];
```
Comment (mirko-pagliai) [inline src/Utility/ThumbCreator.php]: `relative` must be a boolean. `bgcolor` needs a `#`

File: src/Utility/ThumbCreator.php [hunk 2]
```diff
@@ -210,6 +210,32 @@ public function resize($width = null, $heigth = null, array $options = [])
...
+     * Resizes the canvas
+     * @param int $width Required width
+     * @param int $heigth Required heigth
+     * @param array $options Options for the thumbnail
+     * @return \Thumber\Utility\ThumbCreator
+     * @see https://github.com/mirko-pagliai/cakephp-thumber/wiki/How-to-uses-the-ThumbCreator-utility#resizeCanvas
+     * @uses $arguments
+     * @uses $callbacks
+     */
+    public function resizeCanvas($width = null, $heigth = null, array $options = [])
```
Comment (mirko-pagliai) [inline src/Utility/ThumbCreator.php]: `$width` is not optional

**Comment by mirko-pagliai** [comment]
Another thing: missing the same method in the helper class.
I will create a new PR, I do faster.

**Comment by mirko-pagliai** [comment]
Hi @BernardTeske , see PR https://github.com/mirko-pagliai/cakephp-thumber/pull/27
Please, try the code on the [resizeCanvas](https://github.com/mirko-pagliai/cakephp-thumber/tree/resizeCanvas) branch and tell me if it is ok for you, so I can release a new version.

If you have something to add to the code or to tell me, I ask you to do it in the PR linked ;) Thanks

**Comment by mirko-pagliai** [comment]
Released https://github.com/mirko-pagliai/cakephp-thumber/releases/tag/1.3.1

**Comment by BernardTeske** [comment]
Wonderful! I will try!

**Comment by BernardTeske** [comment]
Seams to work very fine! Thank you!

**Pull request closed without merging**